provider/pihole/pihole_test.go

```diff
@@ -51,9 +53,20 @@ func (t *testPiholeClient) deleteRecord(ctx context.Context, ep *endpoint.Endpoi
 		}
 	}
 	t.endpoints = newEPs
+	t.requests.deleteRequests += 1
 	return nil
 }
 
+type requestTracker struct {
+	createRequests int
+	deleteRequests int
```

Comment:
Better unit tests would explicitly specify the expected requests, verifying the dnsname, recordtype, and target.

Reply:
Updated the tests to compare the actual and expected requests instead of just the counts

Reply:
The new version of the tests is asserting that the three new records are created in order, which isn't actually a requirement.

I would have written this to put the expected calls in the tracker. The client would then find the request in the tracker and remove it, asserting that the request was found in the tracker. At the end, the test would assert that there were no remaining expected calls in the tracker.